### PR TITLE
Fix use of messenger in PPOMController

### PR DESCRIPTION
**Pull request by jpuri**
Fixes: MetaMask/MetaMask-planning#1803

Fix use of messenger

### CODE REVIEW

**Comment by jpuri** [comment]
@SocketSecurity ignore bs58/4.0.1
@SocketSecurity ignore ripemd160/2.0.2
@SocketSecurity ignore md5.js/1.3.5
@SocketSecurity ignore @metamask/eth-query/4.0.0
@SocketSecurity ignore rlp/2.2.7
@SocketSecurity ignore @spruceid/siwe-parser/1.1.3
@SocketSecurity ignore  setimmediate/1.0.5

**Comment by jpuri** [comment]
@SocketSecurity ignore bs58/4.0.1
@SocketSecurity ignore ripemd160/2.0.2
@SocketSecurity ignore md5.js/1.3.5
@SocketSecurity ignore @metamask/eth-query/4.0.0
@SocketSecurity ignore rlp/2.2.7
@SocketSecurity ignore @spruceid/siwe-parser/1.1.3
@SocketSecurity ignore  setimmediate/1.0.5

**Comment by jpuri** [comment]
@SocketSecurity@metamask/controller-utils/6.1.0

**Comment by jpuri** [comment]
@SocketSecurity @metamask/controller-utils/6.1.0

**Comment by jpuri** [comment]
@SocketSecurity ignore @metamask/controller-utils/6.1.0

**Comment by naugtur** [comment]
@SocketSecurity ignore @spruceid/siwe-parser@1.1.3
    @SocketSecurity ignore setimmediate@1.0.5
  yes, these packages are published iwth no readme
    
    @SocketSecurity ignore bs58@4.0.1
    @SocketSecurity ignore ripemd160@2.0.2
    @SocketSecurity ignore md5.js@1.3.5
    @SocketSecurity ignore @metamask/eth-query@4.0.0
    @SocketSecurity ignore rlp@2.2.7
  looked up new authors

File: src/ppom-controller.ts [hunk 1]
```diff
@@ -414,6 +419,28 @@ export class PPOMController extends BaseControllerV2<
...
+          PPOMInitialisationStatus.SUCCESS,
+        );
+        this.#checkScheduleFileDownloadForAllChains();
+      })
+      .catch((error: Error) => {
+        this.messagingSystem.publish(
+          'PPOMController:initialisationStateChangeEvent',
+          PPOMInitialisationStatus.FAIL,
+        );
+        console.error(`Error in initialising ppom: ${error.message}`);
```
Comment (digiwand) [inline src/ppom-controller.ts:440]: does PPOM have access to sentry? would it help to capture these errors?

File: src/ppom-controller.ts [hunk 2]
```diff
@@ -304,27 +318,18 @@ export class PPOMController extends BaseControllerV2<
...
-    onNetworkChange(this.#onNetworkChange.bind(this));
 
     // enable / disable PPOM validations as user changes preferences
     onPreferencesChange(this.#onPreferenceChange.bind(this));
 
     // register message handlers
     this.#registerMessageHandlers();
 
+    // subscribe to events
+    this.#subscribeMessageEvents();
```
Comment (digiwand) [inline src/ppom-controller.ts:329]: Q: are there any parts of this lifecycle we should consider unsubscribing to the events?

File: src/ppom-controller.ts (hunk 1 above)
Comment (jpuri) [reply src/ppom-controller.ts:440]: I do not think controllers have direct access to sentry, error from controllers need to be propagated.

File: src/ppom-controller.ts (hunk 2 above)
Comment (jpuri) [reply src/ppom-controller.ts:329]: Good point, un-subscribing should not be needed for network controller.

**Review by segun — APPROVED** [review]

**Pull request merged**